### PR TITLE
Fixes issue 8052: Stackoverflow in toExampleValue() for python client

### PR DESCRIPTION
**Pull request by fbl100**
Added a Set<String> in toExampleValueRecursive() to keep track of which models we have
generated to avoid an infinite recursion for recursive models.  An example of a recursive
model would be a GeoJson GeometryCollection.

@spacether This is a retry of the prior PR that was reverted due to an OS line ending issue.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by spacether** [comment]
Thanks for the new PR! What code did you change to fix the line end issue?
Because there is only one commit I can't tell.

**Comment by fbl100** [comment]
@spacether Yes, that was it.  Sorry, I try to rebase/squash on my side (habit at my day job).

**Comment by spacether** [comment]
Closing and re-opening to try to kick off travis

**Review by spacether — APPROVED** [review]
Thank you for your new PR with the fix!
Merging here and keeping an eye on CI tests...

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java [hunk 1]
```diff
@@ -425,4 +438,31 @@ public void testObjectWithValidations() {
...
+        // uncomment if you need to regenerate the expected value
+        //        PrintWriter printWriter = new PrintWriter("src/test/resources/3_0/issue_8052_recursive_model_expected_value.txt");
+        //        printWriter.write(exampleValue);
+        //        printWriter.close();
+        //        org.junit.Assert.assertTrue(false);
+
+        String expectedValue = Resources.toString(
+                Resources.getResource("3_0/issue_8052_recursive_model_expected_value.txt"),
+                StandardCharsets.UTF_8);
+        expectedValue = expectedValue.replaceAll("\\r\\n", "\n");
```
Comment (spacether) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java:461]: If we kept the expected value in this test and used os separator we would always pass here and wouldn't need the additional file

Comment (fbl100) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java:461]: I'm happy to look into making that change if you'd like.  Are you just suggesting a static final string with the expected value in it?

Comment (spacether) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/python/PythonClientTest.java:461]: Yes static final string. It's fine it was just a thought. No need to do it if tests pass.

**Pull request merged**